### PR TITLE
Splits the RegExp to make it more readable

### PR DESCRIPTION
**Pull request by w4ll3**
This splits the RegExp to make it easier to understand, also allows the nonce to have more then 8 characters. This also fixes the dependabot warning.

To test one could just copy all the fields of the message:
```javascript
const DOMAIN =
	'^(?<domain>([^?#]*)) wants you to sign in with your Ethereum account:\\n';
const ADDRESS = '(?<address>0x[a-zA-Z0-9]{40})\\n\\n';
const STATEMENT = '((?<statement>[^\\n]+)\\n)?\\n';
const URI = '(([^:?#]+):)?(([^?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))';
const URI_LINE = `URI: (?<uri>${URI}?)\\n`;
const VERSION = 'Version: (?<version>1)\\n';
const CHAIN_ID = 'Chain ID: (?<chainId>[0-9]+)\\n';
const NONCE = 'Nonce: (?<nonce>[a-zA-Z0-9]{8,})\\n';
const DATETIME = `([0-9]+)-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(\.[0-9]+)?(([Zz])|([\+|\-]([01][0-9]|2[0-3]):[0-5][0-9]))`;
const ISSUED_AT = `Issued At: (?<issuedAt>${DATETIME})`;
const EXPIRATION_TIME = `(\\nExpiration Time: (?<expirationTime>${DATETIME}))?`;
const NOT_BEFORE = `(\\nNot Before: (?<notBefore>${DATETIME}))?`;
const REQUEST_ID =
	"(\\nRequest ID: (?<requestId>[-._~!$&'()*+,;=:@%a-zA-Z0-9]*))?";
const RESOURCES = `(\\nResources:(?<resources>(\\n- ${URI}?)+))?$`;

const MESSAGE = `${DOMAIN}${ADDRESS}${STATEMENT}${URI_LINE}${VERSION}${CHAIN_ID}${NONCE}${ISSUED_AT}${EXPIRATION_TIME}${NOT_BEFORE}${REQUEST_ID}${RESOURCES}`;
```
And paste it in the Developer tools console and then:
```javascript
new RegExp(MESSAGE)
```

Copy the result and use a website like https://regex101.com/ to validate the regex. A sample message could be:

```
example.com wants you to sign in with your Ethereum account:
0x71C7656EC7ab88b098defB751B7401B5f6d8976F

Example statement for SIWE

URI: https://example.com
Version: 1
Chain ID: 1
Nonce: RJxordNgkIP6JUrh
Issued At: 2021-12-03T13:11:48Z
Expiration Time: 2021-12-05T13:11:48Z
Not Before: 2021-12-04T13:11:48Z
Request ID: some-id
Resources:
- https://example.com/resources/1
- https://example.com/resources/2
```

### CODE REVIEW

File: lib/regex.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
-const DOMAIN = '([^/?#]*)';
-const ADDRESS = '0x[a-zA-Z0-9]{40}';
-const URI = '(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?';
-const DATETIME =
-	'([0-9]+)-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(\\.[0-9]+)?(([Zz])|([\\+|\\-]([01][0-9]|2[0-3]):[0-5][0-9]))';
-const REQUESTID = "[-._~!$&'()*+,;=:@%a-zA-Z0-9]*";
+const DOMAIN =
+	'^(?<domain>([^?#]*)) wants you to sign in with your Ethereum account:\\n';
```
Comment (sbihel) [inline lib/regex.ts:2]: The `^` should probably go in `MESSAGE`

File: lib/regex.ts [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+const VERSION = 'Version: (?<version>1)\\n';
+const CHAIN_ID = 'Chain ID: (?<chainId>[0-9]+)\\n';
+const NONCE = 'Nonce: (?<nonce>[a-zA-Z0-9]{8,})\\n';
+const DATETIME = `([0-9]+)-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(\.[0-9]+)?(([Zz])|([\+|\-]([01][0-9]|2[0-3]):[0-5][0-9]))`;
+const ISSUED_AT = `Issued At: (?<issuedAt>${DATETIME})`;
+const EXPIRATION_TIME = `(\\nExpiration Time: (?<expirationTime>${DATETIME}))?`;
+const NOT_BEFORE = `(\\nNot Before: (?<notBefore>${DATETIME}))?`;
+const REQUEST_ID =
+	"(\\nRequest ID: (?<requestId>[-._~!$&'()*+,;=:@%a-zA-Z0-9]*))?";
+const RESOURCES = `(\\nResources:(?<resources>(\\n- ${URI}?)+))?$`;
```
Comment (sbihel) [inline lib/regex.ts:16]: Same as `^` for `$`

File: lib/regex.ts [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
-const URI = '(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?';
-const DATETIME =
-	'([0-9]+)-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(\\.[0-9]+)?(([Zz])|([\\+|\\-]([01][0-9]|2[0-3]):[0-5][0-9]))';
-const REQUESTID = "[-._~!$&'()*+,;=:@%a-zA-Z0-9]*";
+const DOMAIN =
+	'^(?<domain>([^?#]*)) wants you to sign in with your Ethereum account:\\n';
+const ADDRESS = '(?<address>0x[a-zA-Z0-9]{40})\\n\\n';
+const STATEMENT = '((?<statement>[^\\n]+)\\n)?\\n';
+const URI = '(([^:?#]+):)?(([^?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))';
+const URI_LINE = `URI: (?<uri>${URI}?)\\n`;
```
Comment (sbihel) [inline lib/regex.ts:6]: Maybe, to match the ABNF grammar, the line returns should be moved to the beginning of the following item

**Review by sbihel — APPROVED** [review]

**Comment by sbihel** [comment]
> To test one could just copy all the fields of the message:

You probably know this already, but just to be sure, the parsing tests are run for both the ABNF parser and the regex one.

**Comment by edaxcanim** [comment]
git clone https://github.com/spruceid/siwe
cd siwe/examples/notepad
npm install
npm run dev

**Comment by edaxcanim** [comment]
fix/regex

**Pull request merged**